File: Codigo en 4 lenguajes/Java/TareaPOOReal/clases/EjemploEncapsulacion.java

/*
 * EjemploEncapsulacion.java
 * Presenta 5 ejemplos de encapsulación: clases con atributos privados y sus get/set.
 */

// 1) CuentaBancaria
class CuentaBancaria {
    private String titular;
    private double saldo;
    public CuentaBancaria(String titular, double saldo){ this.titular = titular; this.saldo = saldo; }
    public String getTitular(){ return titular; }
    public void setTitular(String t){ this.titular = t; }
    public double getSaldo(){ return saldo; }
    public void setSaldo(double s){ this.saldo = s; }
}

// 2) Usuario
class Usuario {
    private String username;
    private String password;
    public Usuario(String u, String p){ this.username=u; this.password=p; }
    public String getUsername(){ return username; }
    public void setUsername(String u){ this.username = u; }
    public String getPassword(){ return password; }
    public void setPassword(String p){ this.password = p; }
}

// 3) ProductoEncapsulado
class ProductoEncapsulado {
    private int codigo;
    private String nombre;
    private double precio;
    public ProductoEncapsulado(int codigo, String nombre, double precio){ this.codigo=codigo; this.nombre=nombre; this.precio=precio; }
    public int getCodigo(){ return codigo; } public void setCodigo(int c){ codigo=c; }
    public String getNombre(){ return nombre; } public void setNombre(String n){ nombre=n; }
    public double getPrecio(){ return precio; } public void setPrecio(double p){ precio=p; }
}

// 4) EstudianteEnc
class EstudianteEnc {
    private String nombre; private int edad;
    public EstudianteEnc(String nombre, int edad){ this.nombre=nombre; this.edad=edad; }
    public String getNombre(){ return nombre; } public void setNombre(String n){ nombre=n; }
    public int getEdad(){ return edad; } public void setEdad(int e){ edad=e; }
}

// 5) Orden
class Orden {
    private int id; private boolean pagada;
    public Orden(int id, boolean pagada){ this.id=id; this.pagada=pagada; }
    public int getId(){ return id; } public void setId(int id){ this.id = id; }
    public boolean isPagada(){ return pagada; } public void setPagada(boolean p){ pagada = p; }
}

public class EjemploEncapsulacion {
    public void mostrar() {
        System.out.println("--- Ejemplo: Encapsulación (get/set) ---");
        CuentaBancaria cb = new CuentaBancaria("Camilo", 1500.0);
        System.out.println("Titular: " + cb.getTitular() + ", Saldo: " + cb.getSaldo());
        cb.setSaldo(2000.0); System.out.println("Saldo actualizado: " + cb.getSaldo());

        Usuario u = new Usuario("camilo123","secreto"); System.out.println("Usuario: " + u.getUsername());
        u.setPassword("nuevo"); System.out.println("Password cambiado");

        ProductoEncapsulado pe = new ProductoEncapsulado(10, "Mouse", 25.5);
        System.out.println("Producto: " + pe.getNombre() + ", Precio: " + pe.getPrecio());
        pe.setPrecio(20.0); System.out.println("Precio actualizado: " + pe.getPrecio());

        EstudianteEnc est = new EstudianteEnc("Ana", 21); System.out.println("Estudiante: " + est.getNombre());
        est.setEdad(22); System.out.println("Edad actualizada: " + est.getEdad());

        Orden ord = new Orden(500, false); System.out.println("Orden pagada?: " + ord.isPagada());
        ord.setPagada(true); System.out.println("Orden pagada ahora: " + ord.isPagada());

        System.out.println();
    }
}

File: Codigo de los 4 lenguajes/C#/EjemploEncapsulacion.cs
using System;

public class CuentaBancaria
{
    private string Titular;
    private double Saldo;
    public CuentaBancaria(string titular,double saldo){ Titular=titular; Saldo=saldo; }
    public string GetTitular(){ return Titular; }
    public void SetTitular(string t){ Titular=t; }
    public double GetSaldo(){ return Saldo; }
    public void SetSaldo(double s){ Saldo=s; }
}

public class Usuario
{
    private string Username;
    private string Password;
    public Usuario(string u,string p){ Username=u; Password=p; }
    public string GetUsername(){ return Username; }
    public void SetUsername(string u){ Username=u; }
    public string GetPassword(){ return Password; }
    public void SetPassword(string p){ Password=p; }
}

public class ProductoEncapsulado
{
    private int Codigo;
    private string Nombre;
    private double Precio;
    public ProductoEncapsulado(int codigo,string nombre,double precio){ Codigo=codigo; Nombre=nombre; Precio=precio; }
    public int GetCodigo(){ return Codigo; } public void SetCodigo(int c){ Codigo=c; }
    public string GetNombre(){ return Nombre; } public void SetNombre(string n){ Nombre=n; }
    public double GetPrecio(){ return Precio; } public void SetPrecio(double p){ Precio=p; }
}

public class EstudianteEnc
{
    private string Nombre; private int Edad;
    public EstudianteEnc(string nombre,int edad){ Nombre=nombre; Edad=edad; }
    public string GetNombre(){ return Nombre; } public void SetNombre(string n){ Nombre=n; }
    public int GetEdad(){ return Edad; } public void SetEdad(int e){ Edad=e; }
}

public class Orden
{
    private int Id; private bool Pagada;
    public Orden(int id,bool pagada){ Id=id; Pagada=pagada; }
    public int GetId(){ return Id; } public void SetId(int id){ Id=id; }
    public bool IsPagada(){ return Pagada; } public void SetPagada(bool p){ Pagada=p; }
}

public class EjemploEncapsulacion
{
    public void Mostrar()
    {
        Console.WriteLine("--- Ejemplo: Encapsulación (get/set) ---");

        CuentaBancaria cb = new CuentaBancaria("Camilo",1500.0);
        Console.WriteLine("Titular: " + cb.GetTitular() + ", Saldo: " + cb.GetSaldo());
        cb.SetSaldo(2000.0); Console.WriteLine("Saldo actualizado: " + cb.GetSaldo());

        Usuario u = new Usuario("camilo123","secreto");
        Console.WriteLine("Usuario: " + u.GetUsername());
        u.SetPassword("nuevo"); Console.WriteLine("Password cambiado");

        ProductoEncapsulado pe = new ProductoEncapsulado(10,"Mouse",25.5);
        Console.WriteLine("Producto: " + pe.GetNombre() + ", Precio: " + pe.GetPrecio());
        pe.SetPrecio(20.0); Console.WriteLine("Precio actualizado: " + pe.GetPrecio());

        EstudianteEnc est = new EstudianteEnc("Ana",21);
        Console.WriteLine("Estudiante: " + est.GetNombre());
        est.SetEdad(22); Console.WriteLine("Edad actualizada: " + est.GetEdad());

        Orden ord = new Orden(500,false);
        Console.WriteLine("Orden pagada?: " + ord.IsPagada());
        ord.SetPagada(true); Console.WriteLine("Orden pagada ahora: " + ord.IsPagada());

        Console.WriteLine();
    }
}
